libs/agno/agno/models/openai/realtime.py

```diff
@@ -0,0 +1,270 @@
+from collections.abc import AsyncIterator
+from dataclasses import dataclass
+from os import getenv
+from typing import Any, Dict, Iterator, List, Optional, Union
+
+import httpx
+from pydantic import BaseModel
+
+from agno.exceptions import ModelProviderError
+from agno.media import AudioOutput
+from agno.models.base import Model, MessageData
+from agno.models.message import Message
+from agno.models.response import ModelResponse
+from agno.utils.log import logger
+from agno.utils.openai import add_audio_to_message, add_images_to_message
+
+try:
+    from openai import APIConnectionError, APIStatusError, RateLimitError
+    from openai import AsyncOpenAI as AsyncOpenAIClient
+    from openai import OpenAI as OpenAIClient
+    from openai.types.chat import ChatCompletionAudio
+    from openai.types.chat.chat_completion import ChatCompletion
+    from openai.types.chat.chat_completion_chunk import (
+        ChatCompletionChunk,
+        ChoiceDelta,
+        ChoiceDeltaToolCall,
+    )
+    from openai.types.chat.parsed_chat_completion import ParsedChatCompletion
+except ModuleNotFoundError:
+    raise ImportError("`openai` not installed. Please install using `pip install openai`")
+
+
+@dataclass
+class OpenAIRealtime(Model):
```

Comment:
Should we consider making a whole new base_model for realtime models? They kinda work in fundamentally a different way? It could then support a connection and methods like `send_message`(both text and audio), `process_response`, etc.